interpreter/src/eval/mod.rs

```diff
@@ -44,6 +38,18 @@ impl Evaluator {
     }
   }
 
+  fn eval_block_stmt(&self, block_stmts: Vec<ast::Statement>) -> Option<Object> {
+    let mut result: Option<Object> = None;
+    for stmt in block_stmts {
+      match self.eval_stmt(stmt)? {
+        Object::ReturnValue(val) => return Some(Object::ReturnValue(val)),
```

Comment:
We explicitly don't unwrap the return value and only check the type of each evaluation result. If it's Object::ReturnValue we simply return the Object::ReturnValue, **without unwrapping its value**, so it stops execution in a possible outer block statement and bubbles up to top-level, where it finally gets unwrapped.